libevmasm/AssemblyItem.cpp

```diff
@@ -80,6 +80,10 @@ unsigned AssemblyItem::bytesRequired(unsigned _addressLength) const
 	case PushLibraryAddress:
 	case PushDeployTimeAddress:
 		return 1 + 20;
+	case PushImmutable:
+		return 1 + 32;
+	case AssignImmutable:
+		return -1u; // No upper bound possible, so maximum valus. TODO: check if this can cause overflows anywhere.
```

Comment:
@chriseth Ah yes, this is also something one should carefully look at :-). I'll have a look at all the use cases here. Problem is that ``bytesRequired`` for ``AssignImmutable```depends on the number of occurrences of the immutables (since I have to write once for each occurrence), so it's basically unbounded....

Reply:
Hm - it can cause overflows. Probably only if it occurs in subassembly, but it's still not good...

Reply:
Just use a good estimate. If the number is too large, we have 4-byte jump labels everywhere.

Reply:
I now estimate 1024 occurrences before ``assemble()`` (used as heuristic in the ``PeepholeOptimiser`` only I think) and calculate the precise value during ``assemble()``.